common/src/main/java/com/wynntils/models/containers/BankModel.java

```diff
@@ -83,48 +94,92 @@ public void onScreenClose(ScreenClosedEvent e) {
+        return Optional.ofNullable(pageNamesMap.get(page));
     }
 
     public void saveCurrentPageName(String nameToSet) {
-        Storage<Map<Integer, String>> pageNamesMap = getCurrentNameMap();
+        switch (currentContainer) {
+            case ACCOUNT_BANK -> {
+                customBankPageNames.get().put(currentPage, nameToSet);
+                customBankPageNames.touched();
+            }
+            case BLOCK_BANK -> {
+                customBlockBankPageNames.get().put(currentPage, nameToSet);
+                customBlockBankPageNames.touched();
+            }
+            case BOOKSHELF -> {
+                customBookshelfPageNames.get().put(currentPage, nameToSet);
+                customBookshelfPageNames.touched();
+            }
+            case CHARACTER_BANK -> {
+                Map<Integer, String> nameMap = new TreeMap<>();
 
-        if (pageNamesMap == null) return;
+                if (customCharacterBankPagesNames.get().containsKey(Models.Character.getId())) {
+                    nameMap = customCharacterBankPagesNames.get().get(Models.Character.getId());
+                }
 
-        pageNamesMap.get().put(currentPage, nameToSet);
-        pageNamesMap.touched();
+                nameMap.put(currentPage, nameToSet);
+
+                customCharacterBankPagesNames.get().put(Models.Character.getId(), nameMap);
+                customCharacterBankPagesNames.touched();
+            }
+            case MISC_BUCKET -> {
+                customMiscBucketPageNames.get().put(currentPage, nameToSet);
+                customMiscBucketPageNames.touched();
+            }
+        }
 
         editingName = false;
     }
 
     public void resetCurrentPageName() {
-        Storage<Map<Integer, String>> pageNamesMap = getCurrentNameMap();
-
-        if (pageNamesMap == null) return;
-
-        pageNamesMap.get().remove(currentPage);
-        pageNamesMap.touched();
+        switch (currentContainer) {
+            case ACCOUNT_BANK -> {
+                customBankPageNames.get().remove(currentPage);
+                customBankPageNames.touched();
+            }
+            case BLOCK_BANK -> {
+                customBlockBankPageNames.get().remove(currentPage);
+                customBlockBankPageNames.touched();
+            }
+            case BOOKSHELF -> {
+                customBookshelfPageNames.get().remove(currentPage);
+                customBookshelfPageNames.touched();
+            }
+            case CHARACTER_BANK -> {
+                customCharacterBankPagesNames
+                        .get()
+                        .get(Models.Character.getId())
+                        .remove(currentPage);
+                customCharacterBankPagesNames.touched();
+            }
+            case MISC_BUCKET -> {
+                customMiscBucketPageNames.get().remove(currentPage);
+                customMiscBucketPageNames.touched();
+            }
+        }
```

Comment:
`getCurrentPageMap` could be used here too? Although you are not sure which variable to trigger the save on.. The backside of that method change... Either way `customCharacterBankPagesNames` needs `getOrDefault` since you might produce a NPE there.